### PR TITLE
Update film logic

### PR DESCRIPTION
**Pull request by duc-lt-mor**
# Description
- Implement update film logic

# Evidence

https://github.com/duc-lt-mor/cinema-admin-ui/assets/150450885/fc3e50f2-27db-4dd9-8155-7d707ff32ea6

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] My changes do not affect other functions
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/film/form.tsx [hunk 1]
```diff
@@ -15,23 +15,83 @@ import {
...
+import Image, { StaticImageData } from "next/image";
+
+const FilmForm = (props: { film?: TFilm }) => {
+  const [defaultValues, setDefaultValues] = useState<TFilmFormInput>();
+  const film = useMemo(() => props.film, [props]);
+
+  useEffect(() => {
+    const transformDefaultValues = async () => {
+      if (film) {
+        const newDefaultValues = {
```
Comment (mor-manhnd) [inline app/film/form.tsx:36]: const  newDefaultValues = {...film};
newDefaultValues.genres = ....

File: commons/api-calls.common.ts [hunk 2]
```diff
@@ -50,14 +46,51 @@ export const refreshTokens = async (refreshToken: string) => {
...
-    const result: AxiosResponse<TResponseSuccess<{ message: string }>> =
+    const result: AxiosResponse<TResponse<{ message: string }>> =
       await axiosRef.post(Api.FILM, body, {
         headers: {
           "Content-Type": "multipart/form-data",
         },
       });
 
-    if (result.data) {
+    if (result.data.type === "success") {
```
Comment (mor-manhnd) [inline commons/api-calls.common.ts:56]: should create enum for "success" and error

File: app/film/[id]/page.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { getFilmById } from "@/commons/api-calls.common";
+import FilmForm from "../form";
+import { notFound } from "next/navigation";
+
+const FilmDetails = async ({ params }: { params: { id: string } }) => {
+  const { id: filmId } = params;
+  const result = await getFilmById(filmId);
```
Comment (mor-manhnd) [inline app/film/[id]/page.tsx:7]: Can you use useQuery to fetch API data

Comment (duc-lt-mor) [reply app/film/[id]/page.tsx:7]: no need to use `useQuery` here

**Review by mor-manhnd — APPROVED** [review]
LGTM!

**Pull request merged**